Update links

sethladd opened the pull request:
Will do another pass to the new URLs once redirects are working.

sethladd commented:
cc @Sfshaza @kwalrath

Sfshaza commented:
Hey Seth,

I tested some of these links and the formula isn't right.

For example, you changed this link:

```
http://api.dartlang.org/dart_core/List.html
```

to this:

```
https://api.dartlang.org/dart_core/List.html
```

But when you open that URL, it takes you to the index file, not to the List page:

```
https://api.dartlang.org/1.12.0/index.html
```

The formula that I think we want to use is this:

```
https://api.dartlang.org/stable/dart-core/List-class.html
```

which then redirects you to this:

```
https://api.dartlang.org/1.12.0/dart-core/List-class.html
```

sethladd commented:
Hi @Sfshaza 

What you're seeing is unrelated to this patch. Those redirects were broken anyway. (which, are now fixed, but were not intended to be addressed by this patch)

tldlr: up until 5 min ago, all these redirects were not working, but at least with this patch, we fixed two-three incorrect links and moved everything to SSL.

Starting right-now-now, the directs should be correctly.

Sorry for the conflation.

sethladd commented:
tldr: try again :)

Sfshaza commented:
Confirmed they are working now. Thx!

sethladd commented:
Coolness, thanks for checking!
